src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -12131,5 +12131,512 @@ public string Prop
+                        {
+                            Prop2 = "a";
+                            this = default;
+                            Prop2.ToString(); // expected warning missing
+                            StaticProp.ToString();
+                            return "a";
+                        }
+                    }
+
+                    public string Prop2
+                    {
+                        get => field;
+                        set => field = value;
+                    } = "b";
+
+                    public static string StaticProp
+                    {
+                        get => field;
+                        set => field = value;
+                    } = "c";
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/77991")]
+        public void Nullable_PropInitializer_PropStateAffectedByAssigningStruct()
+        {
+            var source = """
+                #nullable enable
+
+                public struct S()
+                {
+                    public string Prop { get => field; set => field = value; } = "a";
+
+                    public static S StaticProp
+                    {
+                        get => field;
+                        set => field = value;
+                    } = M(
+                        StaticProp = default,
+                        StaticProp.Prop.ToString()); // expected warning missing
+
+                    public static S OtherProp
+                    {
+                        get => field;
+                        set => field = value;
+                    } = M(
+                        StaticProp = default,
+                        StaticProp.Prop.ToString()); // expected warning missing
+
+                    public static S M(S s, string str) => s;
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void Nullable_Cycle_02()
+        {
+            var source = """
+                #nullable enable
+
+                var s = new S
+                {
+                    Resilient = "a", // 1
+                    NonResilient = "b" // 2
+                };
+
+                public struct S
+                {
+                    public S()
+                    {
+                        Resilient.ToString(); // 3
+                        NonResilient.ToString(); // 4
+                    }
+
+                    public S(bool ignored) : this()
+                    {
+                        Resilient.ToString(); // 5
+                        NonResilient.ToString();
+                    }
+
+                    public S((bool, bool) ignored2)
+                    {
+                        this = new();
+                        Resilient.ToString(); // 6
```

Comment:
>Resilient.ToString(); // 6 [](http://example.com/codeflow?start=24&length=26)

Why are we reporting a warning here, rather than initializing the state based on `this = new()`?